### ISSUE TITLE
Add Occupation to Burial fields

### ISSUE THREAD

**Issue by Sandra-AW** [FreeUKGen/MyopicVicar#2737]
Can this field be added in FR so the field can also be added to FreeComETT?

**Comment by stsccfr** [comment]
Vino suggested making a single case for all requests to add new fields to the FlexCSV scheme, so I will add some more here, but it might be a good idea to retitle this one appropriately, if that is possible.  

I request addition of a field for burial records for consecrated_ground, which is found in cemetery registers and which would have the values of Y, yes, N, no, the latter two implying that the register explicitly says unconsecrated ground.

Also for burial records (specifically cemetery registers) a field called 'parish' which would hold either the parish from which the body was removed (specified in earlier registers) or the parish in which the abode was situated (specified in later registers).

Also, fields to hold alternative names for at least the main people in all event types (the person being baptised, buried or groom or bride).  These would allow entering records that say things like:

JONES otherwise SMITH (aliases and adopted names)
Joseph alias Timmy (nicknames)
JONES formerly SMITH formerly WILLIAMS (for women married multiple times)
Helena (Ellen)  (for Catholic registers, which always used Latinised forenames)

Also, suffixes for at least the main people in all event types, to hold things like Junior, Senior, Esquire, III, Bart, etc.

**Comment by Vino-S** [comment]
Occupation for burial record, ready for testing in test3.
Estimate of 5 for other changes

**Comment by stsccfr** [comment]
Vino told me that the new field for occupation in burial records is called 'person_occupation'.  I tested this by uploading a file to test3 and it was accepted.  However, the new field is not being displayed in search results, so a little more needs to be done before this is fully implemented.  (In case it is useful, my test file is called STSCSEBU.CSV and is in account stsccfr on test3. Search for the burial of Enoch Clempson in 1915 and it should show his occupation as Labourer.)

**Comment by DeniseColbert** [comment]
@stsccfr to test if display is correct.

**Comment by stsccfr** [comment]
This is still not working entirely.  I repeated my search for the burial of Enoch CLEMPSON in Staffordshire in 1915.  The search results page still does not show any Occupation field.  If I log into the system as a Coordinator so that I can Browse Entries in the file (STSCSEBU.CSV) then the html table does include a column for Occupation.  However, if I click the SH button to show the full record for Enoch Clempson, the Occupation field is not shown.

**Comment by DeniseColbert** [comment]
Dale to test person_occupation

**Comment by stsccfr** [comment]
I tested this on test3 and it's now working.  The Person Occupation field is displayed in search results.

**Comment by DeniseColbert** [comment]
Dale to test 'consecrated' field

**Comment by stsccfr** [comment]
Yes, the new consecrated_ground field is working.  Files containing it are accepted by the test3 server and it is also being displayed in search results.

**Comment by Vino-S** [comment]
Alternate names: only for forenames

**Comment by stsccfr** [comment]
I'm concerned that we're creating too many CSV fields.  We currently have over 50 fields for marriages.  If we add alternate_name fields and suffix fields for all parties, that could add as many as 24 new fields.  What's that going to look like in FreeComETT or the online editing interface where one has to look through a list of 75 items to pick those one wants?  I think we should discuss the pros and cons and maybe even ask coordinators for their opinions.  It's a non-trivial amount of work to add new fields, and it would be wasted if most transcribers won't bother using them even when they could.

I think the most important reason to add a field is if we are going to make it searchable, because then it becomes a way of finding the record or refining search results.  But the value of a non-searchable field is mainly in transcriber convenience.  If one is transcribing many records that contain information that one has to put into the Notes field, then one naturally wishes for specific fields for that information, because it is faster to enter.  That's fine, but I would want to know how often that situation arises.

**Comment by DeniseColbert** [comment]
Dale to go to Coords for opinions

**Comment by DeniseColbert** [comment]
To consider:

- showing first couple of lines of the notes field on the results page
- making the notes field searchable

**Comment by stsccfr** [comment]
@Vino-S: this is still not working on Test.  If I upload a file containing `relative_occupation`, it is rejected.

**Comment by stsccfr** [comment]
Also need to add support for marriage_by field.  See the FreeComETT repo [#267](https://github.com/FreeUKGen/ComETT/issues/267)
